scripts/logsConversion/logsToJsonConverter.py

```diff
@@ -0,0 +1,398 @@
+                    m=message.strip(),
+                    a=dict_copy(param_dict)
+                ))
+
+            # message or parameters on subsequent entry/entries - add to the last fully formed entry
+            elif line and not self.is_block_table(line) and not self.is_trie_statistics(line):
+                # message in subsequent rows
+                if '===' in line:
+                    self.log_content[-1].m += line.strip()
+
+                # comma separated parameters line on subsequent row (<key>: <value>, <key>: <value>)
+                elif ',' in line and ':' in line:
+                    if self.log_content:
+                        self.add_parameters_for_coma_separatated_entries(self.log_content[-1].a, line.strip())
+
+                # multiple comma separated parameters lines on subsequent rows (<key>=<value>, <key>=<value>)
+                elif ',' in line and '=' in line:
+                    if self.log_content:
+                        param_dict = {}
+                        self.add_parameters_for_coma_separatated_entries(param_dict, line.strip())
+
+                        # if it is not the first parameters row, add a new entry with the same basic values as the last one
+                        if self.log_content[-1].a:
+                            self.log_content.append(self.log_content[-1].copy())
+                        self.log_content[-1].a = param_dict
+
+                # regular parameters line on subsequent entry/entries (<key> = <value> <key> = <value>)
+                elif self.log_content and '=' in line:
+                    self.add_parameters(self.log_content[-1].a, line.strip())
+
+                else:
+                    pass
+                    # irrelevant line.
+
+        # write the no_space_in_message to a file for debugging purposes
+        with open('no_space.txt', 'w') as f:
+            for item in no_space_in_message:
+                f.write(f'{item} => {first_params[no_space_in_message.index(item)]}\n')
+
+    @staticmethod
+    def from_logs(path: str, node_name: str, output_path: str = OUTPUT_FOLDER):
+        result = LogsToJsonConverter(node_name, path, output_path)
+        log_content = load_json(path)
+        result.parse(log_content)
+
+
+def recursive_post_process_keys_for_statistics(txt: str, matches: list[tuple[str, str]], stats_dict: dict[str, str]) -> dict[str, str]:
+    if not matches:
+        return stats_dict
+
+    key, value = matches.pop()
+    key = key.strip()
+    value = value.strip() + (' ' + txt.strip() if txt else '')
+
+    txt = key[:3] if key.startswith('GB ') or key.startswith('MB ') or key.startswith('KB ') else key[:2] if key.startswith('B ') else ''
+    stats_dict[key.replace('GB ', '').replace('MB ', '').replace('KB ', '').replace('B ', '')] = value
+    recursive_post_process_keys_for_statistics(txt, matches, stats_dict)
+    return {k: stats_dict[k] for k in reversed(list(stats_dict.keys()))}
+
+
+def main():
```

Comment:
Can we have this as the first function of the file, then have the others, in "call order" (top-down approach)?